src/terminal/terminal.c

```diff
@@ -777,31 +778,42 @@ void guac_terminal_commit_cursor(guac_terminal* term) {
 
     guac_terminal_char* guac_char;
 
-    guac_terminal_buffer_row* old_row;
-    guac_terminal_buffer_row* new_row;
+    guac_terminal_buffer_row* row;
 
     /* If no change, done */
-    if (term->visible_cursor_row == term->cursor_row && term->visible_cursor_col == term->cursor_col)
+    if (term->cursor_visible && term->visible_cursor_row == term->cursor_row && term->visible_cursor_col == term->cursor_col)
```

Comment:
Will this result in the cursor being repeatedly cleared despite already being invisible?

Reply:
If `cursor_visible == false` we will clear only if `visible_cursor_col != -1 && visible_cursor_row != -1`, then we will unconditionally set those to `-1` which might be redundent but should not be an issue (if we are counting CPU cycles, I dont expect setting those to -1 vs checking if they already are to early out to be noticeably different).

Reply:
Don't worry - no counting of CPU cycles is happening in this case. Orders of magnitude of cycles, sure, but this isn't a performance-critical region of code where individual cycles would have huge multipliers.